cdap-app-fabric/src/main/java/co/cask/cdap/internal/profile/ProfileMetadataPublisher.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.profile;
+
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.common.service.Retries;
+import co.cask.cdap.common.service.RetryStrategies;
+import co.cask.cdap.common.service.RetryStrategy;
+import co.cask.cdap.data2.metadata.writer.MetadataMessage;
+import co.cask.cdap.messaging.MessagingService;
+import co.cask.cdap.messaging.StoreRequest;
+import co.cask.cdap.messaging.client.StoreRequestBuilder;
+import co.cask.cdap.proto.id.EntityId;
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.proto.id.TopicId;
+import com.google.gson.Gson;
+import com.google.inject.Inject;
+
+import java.util.Collections;
+import java.util.Set;
+
+/**
+ * Class for publishing the profile metadata change request to tms
+ */
+public class ProfileMetadataPublisher {
```

Comment:
This isn't really publishing events about profiles. It is also publishing events about entities being created and deleted, but it just happens that the subscriber cares about the impact it has on profiles. Maybe we can name it AdminEventPublisher, as it is actually used whenever some sort of administrative action occurs, like the creation or deletion of entities, or setting preferences.

Reply:
Done